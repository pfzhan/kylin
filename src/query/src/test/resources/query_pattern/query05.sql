--
-- Licensed to the Apache Software Foundation (ASF) under one
-- or more contributor license agreements.  See the NOTICE file
-- distributed with this work for additional information
-- regarding copyright ownership.  The ASF licenses this file
-- to you under the Apache License, Version 2.0 (the
-- "License"); you may not use this file except in compliance
-- with the License.  You may obtain a copy of the License at
--
--     http://www.apache.org/licenses/LICENSE-2.0
--
-- Unless required by applicable law or agreed to in writing, software
-- distributed under the License is distributed on an "AS IS" BASIS,
-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-- See the License for the specific language governing permissions and
-- limitations under the License.
--
select
    c_name,
    c_custkey,
    o_orderkey,
    o_orderdate,
    o_totalprice,
    avg(l_quantity)
from
    v_lineitem
    inner join v_orders on l_orderkey = o_orderkey
    inner join customer on o_custkey = c_custkey
where
    o_orderkey is not null
group by
    c_name,
    c_custkey,
    o_orderkey,
    o_orderdate,
    o_totalprice
having
    avg(l_quantity + 6) > 300
order by
    o_totalprice desc,
    o_orderdate
limit 100