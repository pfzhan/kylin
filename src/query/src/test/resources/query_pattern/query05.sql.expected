SELECT "C_NAME", "C_CUSTKEY", "O_ORDERKEY", "O_ORDERDATE", "O_TOTALPRICE", AVG("L_QUANTITY")
FROM "V_LINEITEM"
INNER JOIN "V_ORDERS" ON ("L_ORDERKEY" = "O_ORDERKEY")
INNER JOIN "CUSTOMER" ON ("O_CUSTKEY" = "C_CUSTKEY")
WHERE ("O_ORDERKEY" IS NOT NULL)
GROUP BY "C_NAME", "C_CUSTKEY", "O_ORDERKEY", "O_ORDERDATE", "O_TOTALPRICE"
HAVING (AVG(("L_QUANTITY" + 6)) > 1)
ORDER BY "O_TOTALPRICE" DESC, "O_ORDERDATE"
LIMIT 1